### PR TITLE
feat: Add option 'trace_root_spans' to instrumentation PG

### PR DESCRIPTION
**Pull request by AS-AlStar**
Add option 'trace_root_spans' to instrumentation PG

### CODE REVIEW

File: instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb [hunk 1]
```diff
@@ -15,6 +15,8 @@ module Patches
         module Connection # rubocop:disable Metrics/ModuleLength
           PG::Constants::EXEC_ISH_METHODS.each do |method|
             define_method method do |*args|
+              return super(*args) unless config[:trace_root_spans] || OpenTelemetry::Trace.current_span.context.valid?
```
Comment (fbogsany) [inline instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb:18]:
```suggestion
              return super(*args) unless config[:trace_root_spans] || OpenTelemetry::Trace.current_span.recording?
```

**Comment by fbogsany** [comment]
@AS-AlStar can you please provide more context on the issue this PR is trying to solve?

**Comment by AS-AlStar** [comment]
> 

I don't want spans to be created on their own. For example, when I make a request to the database in the console "User.last". I spied this option in the redis.
Check in 'rails c', when **trace_root_spans: true**:
`User.last` -  will created span 
when **trace_root_spans: false**:
`User.last` -  don't will created span
but this code generate span:
```
tracer.in_span('operation-name') do |span|
  User.first
end
```

File: instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb (hunk 1 above)
Comment (AS-AlStar) [reply instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb:18]: does this change do the same? I think it's a little different

Comment (fbogsany) [reply instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb:18]: It does the same thing. This is the officially supported way to decide whether to do "expensive" things, based on whether the data will actually be recorded.

**Comment by ahayworth** [comment]
@AS-AlStar @fbogsany I think we should hold off on merging this right now - we discussed this in the SIG today and we feel that this could be accomplished in other ways without needing another config option. We also discussed how our documentation does not really make that obvious or clear, and we will provide examples shortly. 😄

**Comment by arielvalentin** [comment]
@ahayworth do we have docs for this?

**Comment by ericmustin** [comment]
this should stay open i think

**Comment by ericmustin** [comment]
ergh, wrong pr, disregard

**Pull request closed without merging**